tests/h/admin/views/admins_test.py

```diff
@@ -12,102 +12,111 @@
+    def test_add_redirects_to_index_when_user_not_found(self, req):
+        req.params = {"add": "florp"}
 
-        result = views.admins_add(request)
+        result = views.admins_add(req)
 
         assert isinstance(result, httpexceptions.HTTPSeeOther)
         assert result.location == '/adm/admins'
 
-    def test_add_flashes_when_user_not_found(self):
-        request = DummyRequest(params={"add": "florp"})
-        request.session.flash = Mock()
+    def test_add_flashes_when_user_not_found(self, req):
+        req.params = {"add": "florp"}
+        req.session.flash = mock.Mock()
 
-        views.admins_add(request)
+        views.admins_add(req)
 
-        assert request.session.flash.call_count == 1
+        assert req.session.flash.call_count == 1
 
-    def test_remove_makes_users_not_admins(self, users):
-        request = DummyRequest(params={"remove": "cristof"})
+    def test_remove_makes_users_not_admins(self, req, users):
+        req.params = {"remove": "cristof"}
 
-        views.admins_remove(request)
+        views.admins_remove(req)
 
         assert not users['cristof'].admin
 
-    def test_remove_is_idempotent(self, users):
-        request = DummyRequest(params={"remove": "eva"})
+    def test_remove_is_idempotent(self, req, users):
+        req.params = {"remove": "eva"}
 
-        views.admins_remove(request)
+        views.admins_remove(req)
 
         assert not users['eva'].admin
 
-    def test_remove_will_not_remove_last_admin(self, users):
-        views.admins_remove(DummyRequest(params={"remove": "cristof"}))
-        views.admins_remove(DummyRequest(params={"remove": "bojan"}))
-        views.admins_remove(DummyRequest(params={"remove": "agnos"}))
+    def test_remove_will_not_remove_last_admin(self, req, users):
+        req.params = {"remove": "cristof"}
+        views.admins_remove(req)
+        req.params = {"remove": "bojan"}
+        views.admins_remove(req)
+        req.params = {"remove": "agnos"}
+        views.admins_remove(req)
 
         assert users['agnos'].admin
 
-    def test_remove_redirects_to_index(self):
-        request = DummyRequest(params={"remove": "agnos"})
+    def test_remove_redirects_to_index(self, req):
+        req.params = {"remove": "agnos"}
 
-        result = views.admins_remove(request)
+        result = views.admins_remove(req)
 
         assert isinstance(result, httpexceptions.HTTPSeeOther)
         assert result.location == '/adm/admins'
 
-    def test_remove_redirects_to_index_when_user_not_found(self):
-        request = DummyRequest(params={"remove": "florp"})
+    def test_remove_redirects_to_index_when_user_not_found(self, req):
+        req.params = {"remove": "florp"}
 
-        result = views.admins_remove(request)
+        result = views.admins_remove(req)
 
         assert isinstance(result, httpexceptions.HTTPSeeOther)
         assert result.location == '/adm/admins'
 
 
 @pytest.fixture
+def req(db_session):
+    return DummyRequest(db=db_session)
```

Comment:
Oops, sorry -- this *is* `admins_test.py` -- that'll teach me to read PRs on my phone.

Either way, my point stands. It's not a mock, and I fail to see how `req` could be ambiguous in the context of a views test.